apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/common/ClaimDao.java

```diff
@@ -68,6 +78,75 @@ public Object getEntityById(ResourceTypeV2<?> type, String id) {
     return claimEntity;
   }
 
+  public <T> List<T> findAllByMbiHash(
+      Class<T> entityClass, String mbiHash, DateRangeParam lastUpdated) {
+    return findAllByAttribute(entityClass, "mbiHash", mbiHash, lastUpdated);
+  }
+
+  public <T> List<T> findAllByMbi(Class<T> entityClass, String mbi, DateRangeParam lastUpdated) {
+    return findAllByAttribute(entityClass, "mbi", mbi, lastUpdated);
+  }
+
+  @VisibleForTesting
+  <T> List<T> findAllByAttribute(
+      Class<T> entityClass,
+      String attributeName,
+      String attributeValue,
+      DateRangeParam lastUpdated) {
+    List<T> claimEntities = null;
+
+    CriteriaBuilder builder = entityManager.getCriteriaBuilder();
+    CriteriaQuery<T> criteria = builder.createQuery(entityClass);
+    Root<T> root = criteria.from(entityClass);
+
+    criteria.select(root);
+    criteria.where(
+        builder.and(
+            builder.equal(root.get(attributeName), attributeValue),
+            lastUpdated == null
+                ? builder.and()
+                : createDateRangePredicate(root.get("lastUpdated"), lastUpdated, builder)));
+
+    Timer.Context timerClaimQuery = metricRegistry.timer(CLAIM_BY_MBI_METRIC_NAME).time();
+    try {
+      claimEntities = entityManager.createQuery(criteria).getResultList();
+    } finally {
+      long claimByIdQueryNanoSeconds = timerClaimQuery.stop();
+      TransformerUtilsV2.recordQueryInMdc(
+          CLAIM_BY_MBI_METRIC_QUERY,
+          claimByIdQueryNanoSeconds,
+          claimEntities == null || claimEntities.isEmpty() ? 0 : 1);
+    }
+
+    return claimEntities;
+  }
+
+  @VisibleForTesting
+  Predicate createDateRangePredicate(
+      Path<Instant> datePath, DateRangeParam dateRange, CriteriaBuilder builder) {
+    List<Predicate> predicates = new ArrayList<>();
+
+    if (dateRange.getLowerBound() != null) {
+      Instant from = dateRange.getLowerBoundAsInstant().toInstant();
+
+      predicates.add(
+          ParamPrefixEnum.GREATERTHAN.equals(dateRange.getLowerBound().getPrefix())
```

Comment:
should this not also be `ParamPrefixEnum.GREATERTHAN_OR_EQUALS`?

Reply:
if the condition is true, it uses `greaterThan()`, if the condition is false, it uses `greaterThanOrEqualTo()`.  The idea being that the "default" logic would be `greaterThanOrEqualTo()`, since `greaterThan()` requires an exact match.  Inclusive values are typically the default behavior of date ranges for the start date, exclusive for the end date, which is why the upper and lower bound match slightly different conditions.

Reply:
we discussed this on a call and plan on using the functionality defined in [QueryUtils.createLastUpdatePredicate](https://github.com/CMSgov/beneficiary-fhir-data/blob/master/apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/QueryUtils.java#L20)